File: models/Media.js
const mongoose = require('mongoose');

const mediaSchema = new mongoose.Schema({
  type: {
    type: String,
    enum: ['photo', 'video'],
    required: [true, 'Media type is required']
  },
  filePath: {
    type: String,
    required: [true, 'File path is required']
  },
  caption: {
    type: String,
    trim: true
  },
  createdBy: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  }
}, {
  timestamps: true
});

// Index for performance
mediaSchema.index({ type: 1 });
mediaSchema.index({ createdAt: -1 });
mediaSchema.index({ createdBy: 1 });

module.exports = mongoose.model('Media', mediaSchema);

File: mobile/lib/features/media/domain/entities/media.dart
import 'package:equatable/equatable.dart';

class Media extends Equatable {
  final String id;
  final String type; // 'photo' or 'video'
  final String url;
  final String? caption;
  final String? thumbnailUrl;
  final DateTime createdAt;
  final DateTime updatedAt;
  final String? uploadedBy;

  const Media({
    required this.id,
    required this.type,
    required this.url,
    this.caption,
    this.thumbnailUrl,
    required this.createdAt,
    required this.updatedAt,
    this.uploadedBy,
  });

  @override
  List<Object?> get props => [
    id,
    type,
    url,
    caption,
    thumbnailUrl,
    createdAt,
    updatedAt,
    uploadedBy,
  ];
}